### ISSUE TITLE
Add support for response buffering related parameters of Clickhouse HTTP API

### ISSUE THREAD

**Issue by siradjev** [DarkWanderer/ClickHouse.Client#40]
Hi @DarkWanderer, 
is it possible to add `buffer_size` and `wait_end_of_query` parameters into connection string, or add support for wildcard parameters with prefix to cover other and future HTTP API parameters too (e.g. `ch-buffer_size=3000` will pass `buffer_size=3000` setting to Clickhouse API? 

Main reason to use those for us, is that when some query returns initial data quick enough, and the rest of response takes undeterministic amount of time, there is no way to understand whether this slowness comes from query itself/clickhouse, network, application c# code or driver. Additionally, frequently, there is no need to stream response, and it's more straightforward to work on completed result. 

https://clickhouse.tech/docs/en/interfaces/http/#response-buffering
```
You can enable response buffering on the server-side. The buffer_size and wait_end_of_query URL parameters are provided for this purpose.

buffer_size determines the number of bytes in the result to buffer in the server memory. If a result body is larger than this threshold, the buffer is written to the HTTP channel, and the remaining data is sent directly to the HTTP channel.

To ensure that the entire response is buffered, set wait_end_of_query=1. In this case, the data that is not stored in memory will be buffered in a temporary server file.
```

**Comment by DarkWanderer** [comment]
Hi. Sure, that's something I wanted to add, will look into it

**Comment by DarkWanderer** [comment]
Added two options to specify custom HTTP query parameters:

1. Connection string, using `set_` prefix: `Database=localhost;Port=8123;set_buffer_size=5000`
2. `CustomSettings` property of `ClickHouseConnection`: `connection.CustomSettings.Add("buffer_size", "5000")`

Up for next release

**Comment by siradjev** [comment]
Thanks. So I guess that if we add "set_wait_end_of_query=1" in connection string that shall work. 
When do you plan to have next release published in nuget?

**Comment by DarkWanderer** [comment]
Hi @siradjev,

I've released this functionality as part of [2.2.0.279](https://github.com/DarkWanderer/ClickHouse.Client/releases/tag/2.2.0.279)

**Comment by siradjev** [comment]
Thank you very much!
